show/muxcable.py

```diff
@@ -0,0 +1,296 @@
+import json
+import os
+import sys
+
+import click
+import utilities_common.cli as clicommon
+
+from sonic_py_common import multi_asic, device_info
+from swsscommon import swsscommon
+from tabulate import tabulate
+
+platform_sfputil = None
+
+REDIS_TIMEOUT_MSECS = 0
+
+
+# ==================== Methods for initialization ====================
+
+
+# Loads platform specific sfputil module from source
+def load_platform_sfputil():
+    global platform_sfputil
+
+    try:
+        import sonic_platform_base.sonic_sfp.sfputilhelper
+        platform_sfputil = sonic_platform_base.sonic_sfp.sfputilhelper.SfpUtilHelper()
+    except Exception as e:
+        click.echo("Failed to instantiate platform_sfputil due to {}".format(repr(e)))
+        return -1
+
+    return 0
+
+
+#
+# 'muxcable' command ("show muxcable")
+#
+@click.group(name='muxcable', cls=clicommon.AliasedGroup)
+def muxcable():
+    """SONiC command line - 'show muxcable' command"""
+
+    # Load platform-specific sfputil class
+    err = load_platform_sfputil()
+    if err != 0:
+        sys.exit(2)
+
+    # Load port info
+    try:
+        if multi_asic.is_multi_asic():
+            # For multi ASIC platforms we pass DIR of port_config_file_path and the number of asics
+            (platform_path, hwsku_path) = device_info.get_paths_to_platform_and_hwsku_dirs()
+
+            # Load platform module from source
+            platform_sfputil.read_all_porttab_mappings(hwsku_path, multi_asic.get_num_asics())
+        else:
+            # For single ASIC platforms we pass port_config_file_path and the asic_inst as 0
+            port_config_file_path = device_info.get_path_to_port_config_file()
+            platform_sfputil.read_porttab_mappings(port_config_file_path, 0)
+    except Exception as e:
+        log.log_error("Error reading port info (%s)" % str(e), True)
+        sys.exit(3)
+
+    return
+
+
+def get_value_for_key_in_tbl(asic_table, port, key):
+    (status, fvs) = asic_table.get(str(port))
+    if status is not True:
+        click.echo("could not retrieve key {} value for port {} inside table {}".format(key, port, asic_table.getTableName))
+        return None
+    fvp = dict(fvs)
+    value = fvp.get(key, None)
+
+    return value
+
+
+def get_switch_name(config_db):
+    metadata = swsscommon.Table(config_db, 'DEVICE_METADATA')
+    switch_name = get_value_for_key_in_tbl(metadata, "localhost", "hostname")
+    if switch_name is not None:
+        return switch_name
+    return None
+
+
+@muxcable.command()
+@click.argument('port', required=False, default=None)
+@click.option('--json', 'json_flag', required=False, is_flag=True, type=click.BOOL, help="display the output in json format")
```

Comment:
Suggest renaming `json_flag` to `json_output`

Reply:
fixed

Reply:
Doesn't seem to be fixed.

Reply:
fixed